RELEASE.md

```diff
@@ -0,0 +1,17 @@
+# Release process
+
+The release process for CNAO consist on a pair of manual steps:
+1. Create release PR with github action [Create Version PR](https://github.com/kubevirt/cluster-network-addons-operator/actions/workflows/prepare-version.yaml), 
+   it have a pair of fields `versionLevel` with means the semVer version part 
```

Comment:
it has a pair of fields: `versionLevel` and `baseBranch`. `versionLevel` is the semVer version part to bump to and `baseBranch` is the branch to release from.